File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        left = 0
        right = 0
        minimum = 100000000
        subSum = 0
        while right < len(nums):
            subSum = subSum + nums[right]

            while subSum >= target:
                minimum = min(minimum, right - left + 1)
                subSum = subSum - nums[left]
                left += 1

            right += 1    

        return 0 if minimum == 100000000 else minimum

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.dart
class Solution {
  int minSubArrayLen(int target, List<int> nums) {
    
    int left = 0;
    int sum = 0;
    int minLength = nums.length + 1;

     for(int right = 0; right < nums.length ; right++) {
        sum += nums[right];

        while(sum >= target) {
            minLength = min(minLength, (right - left + 1));
            sum -= nums[left];
            left++;
        }
     }
    /// [1,1,1,1,1,1,1,1] t = 11, not meet have to return 0;
    return minLength == nums.length + 1 ? 0 : minLength; 
  }
}